x/auth/vesting/types/vesting_account.go

```diff
@@ -0,0 +1,543 @@
+package types
+
+import (
+	"errors"
+	"time"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	authexported "github.com/cosmos/cosmos-sdk/x/auth/exported"
+	authtypes "github.com/cosmos/cosmos-sdk/x/auth/types"
+	vestexported "github.com/cosmos/cosmos-sdk/x/auth/vesting/exported"
+	"gopkg.in/yaml.v2"
+)
+
+// Assert BaseVestingAccount implements authexported.Account at compile-time
+var _ authexported.Account = (*BaseVestingAccount)(nil)
+
+// Assert vesting accounts implement vestexported.VestingAccount at compile-time
+var _ vestexported.VestingAccount = (*ContinuousVestingAccount)(nil)
+var _ vestexported.VestingAccount = (*PeriodicVestingAccount)(nil)
+var _ vestexported.VestingAccount = (*DelayedVestingAccount)(nil)
+
+// Register the vesting account types on the auth module codec
+func init() {
+	authtypes.RegisterAccountTypeCodec(&BaseVestingAccount{}, "cosmos-sdk/BaseVestingAccount")
+	authtypes.RegisterAccountTypeCodec(&ContinuousVestingAccount{}, "cosmos-sdk/ContinuousVestingAccount")
+	authtypes.RegisterAccountTypeCodec(&DelayedVestingAccount{}, "cosmos-sdk/DelayedVestingAccount")
+	authtypes.RegisterAccountTypeCodec(&PeriodicVestingAccount{}, "cosmos-sdk/PeriodicVestingAccount")
+}
+
+// BaseVestingAccount implements the VestingAccount interface. It contains all
+// the necessary fields needed for any vesting account implementation.
+type BaseVestingAccount struct {
+	*authtypes.BaseAccount
+	OriginalVesting  sdk.Coins `json:"original_vesting" yaml:"original_vesting"`   // coins in account upon initialization
+	DelegatedFree    sdk.Coins `json:"delegated_free" yaml:"delegated_free"`       // coins that are vested and delegated
+	DelegatedVesting sdk.Coins `json:"delegated_vesting" yaml:"delegated_vesting"` // coins that vesting and delegated
+	EndTime          int64     `json:"end_time" yaml:"end_time"`                   // when the coins become unlocked
+	StartTime        int64     `json:"start_time,omitempty" yaml:"start_time,omitempty"`
```

Comment:
Let's keep the original definition.

Reply:
To clarify, you would like `StartTime` and `VestingPeriods` removed on base account, correct? This requires custom 'MarshalYAML` functions to be redefined on `ContinousVestingAccount` and `PeriodicVestingAccount`, otherwise those additional fields will not be rendered.